docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,113 @@
+| Name | string | Never | This required property must be unique to the EdgeX instance and cannot be changed.  The profile name can never be empty and must be unique across the entire EdgeX instance. |
+| Manufacturer | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Description | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Model | string | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include setting the string to an empty string. |
+| Labels | string array | Anytime | As this property does not affect any functionality other than profile query, this property can be changed at any time – to include adding, removing or modifying any of the strings in the array. |
+| Device Resources | array | Situational | New device resources can always be added. Existing device resources can only be removed from the array when the device resource has not yet been associated to any reading or event.|
+| Device Commands | array | Situational | New device commands can always be added.  Existing device commands can only be removed from the array when the device command has not yet been associated to a reading or event.  Please see the device command properties for rules on modifying device commands.Device commands are optional and so the array can always be empty. |
+| *Device Resource Properties* |||
+| Description | string | Anytime | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. |
+| Name | string | Situational | When the device resource is not associated to any reading or event, this field can be modified.  It can never be blank and must be unique across the entire profile. |
+| IsHidden | bool | Anytime | This field allows the device resource to be exposed (default) via the command service.  This change does not affect the behavior but only visibility.  This field is optional but is false by default when not provided. |
+| Tag | string | Anytime | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. Note: adding or removing a tag could effect queries; that is the same query by tag could return different results if a tag is added or removed between query calls.|
+| Properties | Resource Properties | Situational | When the device resource is not associated to any reading or event, this field can be modified.  Because the ValueType and ReadWrite properties of this field are required, it can never be empty. |
```

Comment:
The property should not be changed if there is a Device associated this Profile.